### PR TITLE
Load groups via graph api

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
image

## Related Issue

- Fixes https://github.com/owncloud/web/issues/7554

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-runtime/src/services/auth/userManager.ts [hunk 1]
```diff
@@ -176,18 +176,16 @@ export class UserManager extends OidcUserManager {
...
-      await this.clientService.owncloudSdk.users.getUserGroups(login.id)
-    ])
     this.store.commit('SET_USER', {
       id: login.id,
       uuid: graphUser?.data?.id || '',
       username: login.username || login.id,
       displayname: user.displayname || login['display-name'],
       email: login?.email || user?.email || '',
-      groups: userGroups,
+      groups: graphUser?.data?.memberOf || userGroups,
```
Comment (kulmann) [inline packages/web-runtime/src/services/auth/userManager.ts:188]: Wouldn't it make more sense to set `userGroups = graphUser?.data?.memberOf` inside the `if` block above that deals with graph related requests?

File: packages/web-runtime/src/pages/account.vue [hunk 2]
```diff
@@ -67,15 +67,9 @@
         </dd>
       </div>
       <div class="account-page-info-groups oc-mb oc-width-1-2@s">
-        <dt v-translate class="oc-text-normal oc-text-muted" @click="loadGroups">
```
[the remark this answers is not in the document]
Comment (JanAckermann) [reply packages/web-runtime/src/pages/account.vue:70]: I was wondering too! :D

File: packages/web-runtime/src/services/auth/userManager.ts (hunk 1 above)
Comment (JanAckermann) [reply packages/web-runtime/src/services/auth/userManager.ts:188]: we have the same behavior with the uuid, so I liked to keep that.
But NVM, i can change that

Comment (kulmann) [reply packages/web-runtime/src/services/auth/userManager.ts:188]: hmmm... true. also fine then 🙈

File: packages/web-runtime/src/pages/account.vue [hunk 3]
```diff
@@ -134,18 +126,15 @@ export default defineComponent({
       return null
     },
     groupNames() {
-      return this.groups.join(', ')
+      if (this.capabilities.spaces?.enabled) {
```
Comment (kulmann) [inline packages/web-runtime/src/pages/account.vue:129]: Can't we simply do a type check? if already array of strings -> just join them. If objects -> extract displayName.

Comment (JanAckermann) [reply packages/web-runtime/src/pages/account.vue:129]: https://github.com/owncloud/web/pull/7568/commits/f82c21fcbecfa02ec9a8713c89ab45b1e6359716

To be honest, I am not a fan of duck typing, using the capability makes the code more self-explanatory, especially when the get the graph capability

**Review by kulmann — APPROVED** [review]
LGTM  👍

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/27984/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27984/tracing/re-sharing-alice-2022-9-1-12-25-56.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27984/tracing/re-sharing-brian-2022-9-1-12-26-04.zip 


</p></details>

File: packages/web-runtime/src/pages/account.vue [hunk 4]
```diff
@@ -134,18 +126,15 @@ export default defineComponent({
...
+        return this.user.groups.map((group) => group.displayName).join(', ')
```
Comment (kulmann) [inline packages/web-runtime/src/pages/account.vue:130]: Acceptance tests correctly fail for users without groups. Do this instead:
```suggestion
        return (this.user.groups || []).map((group) => group.displayName).join(', ')
```

Comment (kulmann) [reply packages/web-runtime/src/pages/account.vue:130]: ah, nice, yours is even better 👍

Comment (JanAckermann) [reply packages/web-runtime/src/pages/account.vue:130]: <3

**Pull request merged**